### ISSUE TITLE
Windows build failures and v0.149.0 release status

### ISSUE THREAD

**Issue by mroch** [facebook/flow#8645]
The master branch and the [v0.149.0 release](https://github.com/facebook/flow/releases/tag/v0.149.0) currently fail to build on Windows, which is holding up the release of `flow-bin@0.149.0`.

We believe this is due to the way OCaml dynlink works under cygwin, which finally decided to bite us in binutils 2.36+ (https://github.com/alainfrisch/flexdll/issues/50, cc @dra27). 

We use dynlink for `ppx_deriving` plugins, which now fail with errors like this:

```
Error: Cannot load ppx_deriving.eq: error loading shared library:
       Dynlink.Error (Dynlink.Cannot_open_dll "Failure(\"flexdll error:
       cannot relocate RELOC_REL32, target is too far: 00000000db79cc02 
       ffffffffdb79cc02\")")
```

I'm working to downgrade our CI back to binutils 2.35 temporarily to unblock the release. We'll package all our ppx's into one static binary to avoid dynlink longer term.

**Comment by dra27** [comment]
This is (almost) correct - we'd also spotted it too and the fix is in https://github.com/ocaml/ocaml/pull/10351 (it'll be in 4.12.1/4.13.0).

The part that is not correct is that this has nothing to do with Cygwin, although as it happens Cygwin already behaved the same way. The issue is that binutils 2.36 intentionally starts placing image base addresses at 4GiB+ for x86_64 which is therefore picked up by `x86_64-w64-mingw32-ld` and it appears to putting DLLs and EXEs more than 2GiB apart (it's not clear if that's intentional - on 64-bit Cygwin, that _is_ intentionally done).

**Comment by dra27** [comment]
There is a dirtier patch needed to flexlink to allow older versions of OCaml to continue working, but I haven't written it yet...

**Comment by mroch** [comment]
Update: #8646 downgrades our Circle builds to binutils 2.35. the Windows artifact attached to the 0.149.0 release and included in `flow-bin@0.149.0` was generated by CircleCI from that PR.
